### PR TITLE
add git annex

### PR DESCRIPTION
**Pull request by notestaff**
This PR adds only a Linux recipe for git-annex.  That recipe works fine (the full git-annex test suite passes).  The MacOS recipe is proving more difficult.  I'll keep working on that, but in the meantime it'd be useful to have git-annex in conda-forge at least for Linux.

### CODE REVIEW

**Comment by notestaff** [comment]
getting git-annex into conda-forge would help with PR #1720

**Comment by notestaff** [comment]
this PR also includes recipes for rsync and popt, same as PR #4150 , as these dependencies of git-annex are not yet in conda-forge.

**Comment by yarikoptic** [comment]
woohoo!
Kudos @notestaff for this work!
If needed, I bet we could ask @joeyh (git-annex author) to advice on particular aspects of building for OSX (and Windows... he was fixing up Windows build recently)

**Comment by notestaff** [comment]
@conda-forge/core Would you consider merging this without a working osx build, or is that a requirement?

**Comment by notestaff** [comment]
No, this is using ghc installed by Stack .

**Comment by jakirkham** [comment]
Building on Linux alone is fine. Would raise an issue on the feedstock for adding macOS.

File: recipes/git-annex/build.sh [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+mkdir -p $PACKAGE_HOME
+
+export CPPFLAGS=-I${PREFIX}/include
+export CFLAGS=-I${PREFIX}/include
+export CXXFLAGS=-I${PREFIX}/include
+export CPPFLAGS=-I${PREFIX}/include
+export C_INCLUDE_PATH=${PREFIX}/include
+export CPLUS_INCLUDE_PATH=${PREFIX}/include
+export LDFLAGS=-L${PREFIX}/lib
+export LIBRARY_PATH=${PREFIX}/lib
```
Comment (isuruf) [inline recipes/git-annex/build.sh]: Do you need the env variables above? `toolchain` sets some of these.

Comment (notestaff) [reply recipes/git-annex/build.sh]: I've tried removing some of these, got errors -- see earlier commits.

**Comment by yarikoptic** [comment]
FWIW, if this one is merged, it would remove the road-block toward #1720 as well (if only on Linux to start with)

File: recipes/git-annex/build.sh (hunk 1 above)
Comment (isuruf) [reply recipes/git-annex/build.sh]: Then you should append instead of setting the env variables

**Comment by notestaff** [comment]
@isuruf I removed all but one setting of environment variables in build.sh , and made that one (LIBRARY_PATH) append to any existing path.  Also updated to latest version of git-annex.  Let me know if you see other issues.

File: recipes/git-annex/build.sh (hunk 1 above)
Comment (notestaff) [reply recipes/git-annex/build.sh]: @isuruf this has been fixed.

**Comment by luispedro** [comment]
Building Haskell packages on Mac OS X is an issue that bioconda also faces:

https://github.com/bioconda/bioconda-recipes/issues/6607

I would imagine that whenever someone finds a robust solution, it'd apply to all of these packages.

**Comment by luispedro** [comment]
Also, +1 on adding git-annex: I would love to be able to install it with `conda`.

**Comment by luispedro** [comment]
Any reason this hasn't been merged?

**Comment by yarikoptic** [comment]
I wish i could do that ;-)

**Comment by dougalsutherland** [comment]
Sorry for the slow movement on this one, especially after you did so much work @notestaff.

I just:
- rebased onto `master`, since the CI scripts changed in the meantime
- updated some pinnings to [the preferred versions](https://github.com/conda-forge/conda-forge-pinning-feedstock/blob/master/recipe/conda_build_config.yaml)
- updated to the latest version.

Assuming Circle passes, it then looks good to me.

**Comment by notestaff** [comment]
@dougalsutherland thanks a lot!

**Comment by dougalsutherland** [comment]
Merged! Thanks for all your work on this @notestaff.

**Comment by yarikoptic** [comment]
awesome -- thank you @notestaff and @dougalsutherland for making it happen!

**Pull request merged**